pkg/astnormalization/implicit_extend_root_operation.go

```diff
@@ -0,0 +1,57 @@
+package astnormalization
+
+import (
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+)
+
+/*
+type Query {...}
+type Mutation {...}
+type Subscription {...}
+
+will be,
+
+extend type Query {...}
+extend type Mutation {...}
+extend type Subscription {...}
+
+this also works if root types are defined in schema{...} with other names.
+root types are left unmodified if they have no fields, directives or implements any interface.
+*/
+type implicitExtendRootOperationVisitor struct {
+	operation *ast.Document
+}
+
+func implicitExtendRootOperation(walker *astvisitor.Walker) {
+	v := &implicitExtendRootOperationVisitor{}
+	walker.RegisterEnterDocumentVisitor(v)
+	walker.RegisterEnterObjectTypeDefinitionVisitor(v)
+}
+
+func (v *implicitExtendRootOperationVisitor) EnterDocument(operation, _ *ast.Document) {
+	v.operation = operation
+}
+
+func (v *implicitExtendRootOperationVisitor) EnterObjectTypeDefinition(ref int) {
+	node := v.operation.ObjectTypeDefinitions[ref]
+	if !(node.HasDirectives || node.HasFieldDefinitions || len(node.ImplementsInterfaces.Refs) > 0) {
```

Comment:
@spetrunin majorly because of these two lines
https://github.com/TykTechnologies/graphql-go-tools/blob/master/pkg/federation/sdlmerge/sdlmerge.go#L33
https://github.com/TykTechnologies/graphql-go-tools/blob/master/pkg/federation/sdlmerge/object_type_extending.go#L32